ce/ce/session.ts

```diff
@@ -125,139 +118,31 @@ export class Session {
-    // normalize the location first.
-
-    registryLocation = typeof registryLocation === 'string' ? await this.parseLocation(registryLocation) || this.parseUri(registryLocation) : registryLocation;
-
-    if (!registryLocation) {
-      return undefined;
-    }
-
-    // if the registry from that location is already loaded, return it.
-    const r = this.registries.getRegistry(registryLocation.toString());
-    if (r) {
-      return r;
-    }
-
-    // not already loaded
-    registryLocation = this.parseUri(registryLocation);
-
-    switch (registryKind) {
-
-      case 'artifact':
-        switch (registryLocation.scheme) {
-          case 'https':
-            return this.registries.add(new RemoteRegistry(this, registryLocation));
-
-          case 'file':
-            return this.registries.add(new LocalRegistry(this, registryLocation));
-
-          default:
-            throw new Error(i`Unsupported registry scheme '${registryLocation.scheme}'`);
-        }
+  parseLocation(location: string): Uri {
+    // Drive letter, absolute Unix path, or drive-relative windows path, treat as a file
+    if (/^[A-Za-z]:/.exec(location) || location.startsWith('/') || location.startsWith('\\')) {
+      return this.fileSystem.file(location);
     }
-    throw new Error(i`Unsupported registry kind '${registryKind}'`);
-  }
-
-  async isLocalRegistry(location: Uri | string): Promise<boolean> {
-    location = this.parseUri(location);
-
-    if (location.scheme === 'file') {
-      if (await isIndexFile(location)) {
-        return true;
-      }
 
-      if (await location.isDirectory()) {
-        const index = location.join(registryIndexFile);
-        if (await isIndexFile(index)) {
-          return true;
-        }
-        const s = this;
-        let result = false;
-        const q = new Queue();
-
-        // still could be a folder of artifact files
-        // eslint-disable-next-line no-inner-declarations
-        async function process(folder: Uri) {
-          for (const [entry, type] of await folder.readDirectory()) {
-            if (result) {
-              return;
-            }
-
-            if (type & FileType.Directory) {
-              await process(entry);
-              continue;
-            }
-
-            if (type & FileType.File && entry.path.endsWith('.json')) {
-              void q.enqueue(async () => { result = result || await isMetadataFile(entry, s); });
-            }
-          }
-        }
-        await process(location);
-        await q.done;
-        return result; // whatever we guess, we'll use
-      }
-      return false;
+    // Has a scheme, it's a URI
+    if (/^(\w+):/.exec(location)) {
+      return this.fileSystem.parse(location);
     }
 
-    return false;
-  }
-
-  async isRemoteRegistry(location: Uri | string): Promise<boolean> {
-    return this.parseUri(location).scheme === 'https';
+    // Relative path
+    return this.currentDirectory.join(location);
```

Comment:
Is this making URIs relative to the file they're found in, the top level file, or the current working directory?

Reply:
Current working directory. Hmmmm.... maybe we shouldn't have that fallback and fix console arguments to be parsed with something else.

Reply:
OK, removed CWD fallback here, audited all callers and verified that they never expected relative paths, and also audited all callers of `filesystem.file()` and added Path.resolve calls to them where reasonable.